generator/test_case_generator.go

```diff
@@ -60,28 +63,34 @@ func genMatrix(targetOS string, testDirList []string) []map[string]string {
 	var testMatrix []map[string]string
 	err = json.Unmarshal(byteValueTestMatrix, &testMatrix)
 	if err != nil {
-		log.Panicf("can't unmarshall file %v_test_matrix.json err %v", targetOS, err)
+		log.Panicf("can't unmarshall file %v_test_matrix.json err %v", testType, err)
 	}
 
 	var testMatrixComplete []map[string]string
 	for _, test := range testMatrix {
 		for _, testDirectory := range testDirList {
 			testLine := copyMap(test)
 			testLine[testDir] = testDirectory
+			// In golang marshal is in order. This will add test type, test dir, and os name to the front of the map
+			// The github action test cuts off the full test name
+			// This means we will always show the test type, test dir and os in the test name
+			testLine[prefix+testDir] = testDirectory
+			testLine[prefix+os_name] = testLine[os_name]
+			testLine[prefix+testTypeName] = testType
 			testMatrixComplete = append(testMatrixComplete, testLine)
```

Comment:
Rather than introducing a weird `1_` prefix, we could make this a struct and implement our own marshaller to preserve order.